src/intertypes/InterTypes.jl

```diff
@@ -100,6 +100,12 @@ end
 
 function intertype end
 
+struct ExportTarget{type} end
```

Comment:
The more idiomatic way to handle this would be to make a generic type and a singleton subtype for each concrete ExportTarget.

```julia
abstract type ExportTarget end
struct PythonTarget <: ExportTarget end
struct JSONTarget <: ExportTarget end
```

That way you can check for the exhaustive list of valid export targets with `subtypes(ExportTarget)`. You can also do a hierarchical scheme like distinguishing between programming language output formats and serialization output formats.

```julia
abstract type ExportTarget end
abstract type CodeTarget <: ExportTarget end
abstract type SerializationTarget <: ExportTarget end

struct PythonTarget <: CodeTarget end
struct RustTarget <: CodeTarget end
struct HaskellTarget <: CodeTarget end

struct JSONTarget <: SerializationTarget end
struct BSONTarget <: SerializationTarget end
struct ProtoBufTarget <: SerializationTarget end
```

Reply:
@jpfairbanks My only question here would be what would be the best way to actually call it?

I could 
1. provide the ExportTarget subtypes to the user. This _feels_ more idiomatic but it's also a lot to import and seems a bit more complicated to the end user. The different targets are also more discoverable.
2. map symbols to the ExportTarget. The user would just provide the symbols and an ugly dictionary would map symbols to the right type.

Typing this out (1) seems like the way to go.

Reply:
The user only needs to import the types for the formats they are using. So that's definitely the way to go. Sounds like you got there by rubber ducking it to GitHub.